addons/rating/models/rating.py

```diff
@@ -106,9 +106,11 @@ def _find_parent_data(self, values):
             'parent_res_id': False,
         }
         if hasattr(current_record, 'rating_get_parent'):
-            parent_res_model = getattr(current_record, current_record.rating_get_parent())
-            data['parent_res_model_id'] = self.env['ir.model']._get(parent_res_model._name).id
-            data['parent_res_id'] = parent_res_model.id
+            field = current_record.rating_get_parent()
```

Comment:
With something like
```
if getattr(current_record, 'rating_get_parent', None):
    blahblah
```
You will need only 1 level of if :) . Indeed it will test both the existence of the method and that it returns anything. You could also check that field is in record to be sure.

Reply:
Hello @tde-banana-odoo 
In this case ``getattr(current_record, current_record.rating_get_parent())`` the method ``rating_get_parent()`` of ``rating.rating`` model return ``None`` and make issue.

``getattr()`` failed due to ``None`` in second parameter.

So here, we need to check the result of ``current_record.rating_get_parent()`` and not pass in ``getattr`` if Its ``None``

Reply:
That's not what I said. I think something like
```
field_name = getattr(current_record, 'rating_get_parent', None)
if field_name:
    parent_res_model = current_record[field_name]
```
should do  the trick. Could you try ?